File: April 2020/21 - Leftmost Column with at Least a One/Solution.py
class Solution:
    def leftMostColumnWithOne(self, binaryMatrix: 'BinaryMatrix') -> int:
        n, m = binaryMatrix.dimensions()
        leftmost = right = m
        
        for i in range(n):
            left = 0
            while left < right:
                mid = left + (right - left) // 2
                if(binaryMatrix.get(i, mid)):
                    right = mid
                else:
                    left = mid + 1
            leftmost = left
        
        return -1 if leftmost == m else leftmost

File: April 2020/21 - Leftmost Column with at Least a One/Solution.cpp
class Solution {
public:
    int leftMostColumnWithOne(BinaryMatrix &bM) {
        vector<int> dim = bM.dimensions();
        int n = dim[0], m = dim[1];
        int i, j = m;
        int rptr = m, lptr = 0;
        
        for(i = 0; i < n; ++i) {
            lptr = 0;
            // binary search
            while(lptr < rptr) {
                int mid = lptr - (lptr - rptr) / 2;
                if(bM.get(i, mid))
                    rptr = mid;
                else
                    lptr = mid + 1;
            }
            // assign leftmost column with one to j
            j = lptr;
        }
        
        return j == m ? -1 : j;
    }
};